src/URI_TEST.cc

```diff
@@ -890,8 +890,8 @@ TEST(URITEST, HasAuthority)
     // Modifyng path updates string
     uri.Path() = URIPath("new_authority.com/another/path");
 
-    EXPECT_EQ("new_authority.com/another/path/", uri.Path().Str());
-    EXPECT_EQ("https://new_authority.com/another/path/", uri.Str());
+    EXPECT_EQ("new_authority.com/another/path", uri.Path().Str());
+    EXPECT_EQ("https://new_authority.com/another/path", uri.Str());
```

Comment:
I don't understand why these expectations would change :thinking:

Reply:
Ah yes, I forgot that I did this.  I don't understand, either...

Reply:
I found out why.  By using the automatically generated functions, we get one that copies all of the members.

The issue is that the `URIPath::Implementation` struct has a `trailingSlash` member: https://github.com/ignitionrobotics/ign-common/blob/ign-common4/src/URI.cc#L54-L73

The originally written copy-assignment operator doesn't copy this member:
https://github.com/ignitionrobotics/ign-common/blob/ign-common4/src/URI.cc#L586-L591

And the test exercises this copy-assignment: 
https://github.com/ignitionrobotics/ign-common/blob/ign-common4/src/URI_TEST.cc#L890-L894

In this case, `ImplPtr` has shown that the previous behavior is actually a bug, and we should fix it in previous versions.

Reply:
Oh, all these refs were to ign-common4, but the same is true on main

Reply:
Ohhhh so this PR includes more than QOL, it's also a bugfix! Nice detective work :+1: